### DISCUSSION TITLE
Debugging and Testing the project

### DISCUSSION THREAD

**Discussion by samsunlu55atakum** [unoplatform/uno#13399] [Q&A]
Hello everyone! I will try to migrate my WPF project which is already done to UNO Platform for the app will run on linux. I will need to solve some bug, test the libraries that can run on WPF but cannot in UNO, some classes will not work on Uno. So, I need to change or integrate to some code. So, I don't know how to handle these bugs. Actually, I did not even debug in .Net before. How can I handle these bugs and where should I start on it?

**Comment by MarkBrainfire** [comment]
https://avaloniaui.net/

**Reply by samsunlu55atakum** [reply]
I want to use Uno, not avalonia.

**Comment by MarkBrainfire** [comment]
They've released v 11 and have a product XPF which is advertized to take a WPF project to Linux.

I thought you ought to know. 

I too am using Uno because I am creating a substantial project in WinUI 3.

Good luck with your project.  ---- On Fri, 25 Aug 2023 03:54:01 -0500  ***@***.***  wrote ----
I want to use Uno, not avalonia.

**Reply by samsunlu55atakum** [reply]
I noticed xpf while searching for alternative techs. Thanks.

**Comment by jeromelaban** [comment]
@samsunlu55atakum thanks for asking the question. Could you tell more about the classes you want to use that are not available?

**Reply by samsunlu55atakum** [reply]
I am gonna try run my wpf app on WinUI. After that, I will use Uno Platform to run my wpf app on linux. So, Actually, I am new to use these techs. I am just asking for if there will some bugs, incompatible api's and libraries. I mean, I have not dive into this subject yet. I am just want to know these library and api's. Thanks for answering @jeromelaban
